lib/functions.php

```diff
@@ -174,6 +177,26 @@ function cryptoEnable($socket, array $options = []) {
         }
         if (empty($options["cafile"])) {
             $options["cafile"] = __DIR__ . "/../var/ca-bundle.crt";
+
+            if (class_exists("Phar") && !empty(Phar::running(true))) {
+                // Legacy PHP 5.5 needs bundle external from Phar,
+                // because we verify peers manually in amphp/socket.
+                // Yes, this is blocking but way better than just an error.
+                if (!isset($__amphp_artax_openssl_ca_bundle_file)) {
+                    $bundle = __DIR__ . "/../var/ca-bundle.crt";
+                    $bundleContent = file_get_contents($bundle);
+                    $__amphp_artax_openssl_ca_bundle_file = tempnam(sys_get_temp_dir(), "openssl-ca-bundle-");
+                    file_put_contents($__amphp_artax_openssl_ca_bundle_file, $bundleContent);
+
+                    register_shutdown_function(function() use ($__amphp_artax_openssl_ca_bundle_file) {
+                        if (file_exists($__amphp_artax_openssl_ca_bundle_file)) {
+                            unlink($__amphp_artax_openssl_ca_bundle_file);
```

Comment:
The file_exists() is redundant (especially because race conditions). To be safe, just use `@unlink()`.